crates/cxx-qt-gen/src/generator/cpp/property/meta.rs

```diff
@@ -7,12 +7,22 @@ use crate::generator::naming::property::QPropertyNames;
 
 /// Generate the metaobject line for a given property
 pub fn generate(idents: &QPropertyNames, cxx_ty: &str) -> String {
-    format!(
-        "Q_PROPERTY({ty} {ident} READ {ident_getter} WRITE {ident_setter} NOTIFY {ident_notify})",
+    let mut output: String = format!(
+        "Q_PROPERTY({ty} {ident} READ {ident_getter} ",
         ty = cxx_ty,
         ident = idents.name.cxx_unqualified(),
-        ident_getter = idents.getter.cxx_unqualified(),
-        ident_setter = idents.setter.cxx_unqualified(),
-        ident_notify = idents.notify.cxx_unqualified()
-    )
+        ident_getter = idents.getter.cxx_unqualified()
+    );
+    // Write
+    if let Some(name) = &idents.setter {
+        output += format!("WRITE {} ", name.cxx_unqualified()).as_str();
+    }
+    // Notify
+    if let Some(name) = &idents.notify {
+        output += format!("NOTIFY {}", name.cxx_unqualified()).as_str();
+    }
+
+    output += ")";
+
+    output
```

Comment:
wonder if it is better or worse to build up a vec of the inner parts then write at the end joining them, rather than modifying a string?

eg (could have better naming but you get the idea)

```rust
let mut parts = vec![format!("READ {ident_getter}")];

// if write is enabled
parts.push(format!("WRITE {}", name.cxx_unqualified());

format!("Q_PROPERTY({ty} {ident} {meta_parts})", meta_parts = parts.join(" "))
```

Reply:
done